### DISCUSSION TITLE
Default Headers for Email

### DISCUSSION THREAD

**Discussion by nie7321** [laravel/framework#50202] [Ideas]
Is there currently a way to configure some default email headers for all emails from an app, similar to setting the `config/mail.php` from address/name?

This would be pretty handy for something like Mailpit serving a couple in-development applications: you can include an `X-Tags` header and it will sort the emails it captures by application.

Since you can send mail in a few ways (mailables, `Mail::to(...)->...`, notifications), it seems like this might be something that would be easiest to manage in the config.

Thoughts? I don't want to try to implement something if somebody has a good one-liner for a service provider that solves this perfectly 😅

**Comment by henzeb** [comment]
Something like this, but with headers? 

https://laravel.com/docs/10.x/mail#using-a-global-from-address

**Reply by nie7321** [reply]
Yeah, exactly. It would probably need to look more like this because you can have a couple:

```php
config/mail.php

return [
    // . . .

    'headers' => [
        // Since you can have multiple headers, it's gotta be an array.
        // For my Mailpit use-case, I'd add this:
       'X-Tags' => env("APP_NAME"),
    ],

**Comment by robjbrain** [comment]
+1 to this.

It would be very useful for ignoring auto responders for example.

Symfony recommends adding the following to suppress auto responders https://symfony.com/doc/current/mailer.html#message-headers 

```
->addTextHeader('X-Auto-Response-Suppress', 'OOF, DR, RN, NRN, AutoReply')
```

The current config seems to be used via this `setGlobalAddress` method https://github.com/illuminate/mail/blob/5a43f89030f3fa7ac1cd7da8a48b2ad7428b61c0/MailManager.php#L474

Weirdly https://github.com/illuminate/mail/blob/master/Mailer.php does not seem to reference headers in anyway.

I think @createSymfonyTransport https://github.com/illuminate/mail/blob/5a43f89030f3fa7ac1cd7da8a48b2ad7428b61c0/MailManager.php#L152 would be the best place for it.

If you look at the AWS SES one it accepts the entire `$config` which could contain default headers https://github.com/illuminate/mail/blob/5a43f89030f3fa7ac1cd7da8a48b2ad7428b61c0/MailManager.php#L244

But it's all a little too opaque for someone like me to try and submit a pull request and would likely be a very big change in order to satisfy the vast range of transports available.

**Comment by robjbrain** [comment]
In the mean time if you are using mailables you could just have a BaseMailable that all your other mailables extend.

```
namespace App\Mail;

use Illuminate\Mail\Mailable;
use Illuminate\Mail\Mailables\Headers;

class BaseMailable extends Mailable
{
    public function headers(): Headers
    {
        return new Headers(
            text: [
                'X-Auto-Response-Suppress' => 'OOF, DR, RN, NRN, AutoReply',
            ]
        );
    }
}
```
